src/MongoDB.Driver.Core/Core/Bindings/CoreSession.cs

```diff
@@ -434,17 +435,41 @@ private void EnsureStartTransactionCanBeCalled()
         {
             if (_currentTransaction == null)
             {
-                return;
+                EnsureTransactionsAreSupported();
             }
+            else
+            {
+                switch (_currentTransaction.State)
+                {
+                    case CoreTransactionState.Aborted:
+                    case CoreTransactionState.Committed:
+                        break;
 
-            switch (_currentTransaction.State)
+                    default:
+                        throw new InvalidOperationException("Transaction already in progress.");
+                }
+            }
+        }
+
+        private void EnsureTransactionsAreSupported()
+        {
+            var connectedDataBearingServers = _cluster.Description.Servers.Where(s => s.State == ServerState.Connected && s.IsDataBearing).ToList();
+
+            if (connectedDataBearingServers.Count == 0)
             {
-                case CoreTransactionState.Aborted:
-                case CoreTransactionState.Committed:
-                    return;
+                throw new NotSupportedException("StartTransaction cannot determine if transactions are supported because there are no connected servers.");
+            }
 
-                default:
-                    throw new InvalidOperationException("Transaction already in progress.");
+            foreach (var connectedDataBearingServer in connectedDataBearingServers)
+            {
+                if (connectedDataBearingServer.Type == ServerType.ShardRouter)
```

Comment:
Does this logic cover all cases from the DRIVER ticke?t:

>startTransaction SHOULD report an error if the driver can detect that transactions are not supported by the deployment. A deployment does not support transactions when the deployment does not support sessions, or maxWireVersion < 7, or the topology type is Sharded

>in particular `maxWireVersion` ?

what about checking `session` supporting by these rules: https://github.com/mongodb/specifications/blob/master/source/sessions/driver-sessions.rst#how-to-check-whether-a-deployment-supports-sessions
?
In particular, I'm not sure about the 2nd rule:
>2. Having verified in step 1 that the TopologyDescription includes at least one connected server a driver can now determine whether sessions are supported by inspecting the TopologyType and logicalSessionTimeoutMinutes property.